packages/amazon/src/serverGroup/details/scalingPolicy/upsert/alarm/DimensionsEditor.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import { flatMap, uniq} from 'lodash';
+import { Subject } from 'rxjs';
+import { IScalingPolicyAlarmView } from '../../../../../domain';
+import { CloudMetricsReader, IMetricAlarmDimension, IServerGroup, ReactSelectInput, TextInput, useData } from '@spinnaker/core';
+
+import './dimensionsEditor.less';
+
+export interface IDimensionsEditorProps {
+  alarm: IScalingPolicyAlarmView;
+  namespaceUpdated?: Subject<void>;
+  serverGroup: IServerGroup;
+  updateAvailableMetrics: () => void;
+}
+
+export const DimensionsEditor = ({
+  alarm,
+  serverGroup,
+  updateAvailableMetrics
+}: IDimensionsEditorProps) => {
+ const [ dimensions, setDimensions ] = React.useState(alarm.dimensions || []);
+
+  const fetchDimensions = () => {
+    return CloudMetricsReader.listMetrics('aws', serverGroup.account, serverGroup.region, { namespace: alarm.namespace }).then(results => {
+      const allDimensions = flatMap(results, (r) =>  r.dimensions);
+      const sortedDimensions = uniq(allDimensions.map(d => d.name)).sort();
+      return sortedDimensions;
+    });
+  };
+  const { result: dimensionOptions } = useData(fetchDimensions, [], [alarm.namespace, serverGroup.name]);
+
+  const addDimension = () => {
+    const newDimensions = [...dimensions, {} as IMetricAlarmDimension];
+    alarm.dimensions = newDimensions;
+    setDimensions(newDimensions);
```

Comment:
Because the parent components are still in angular, the dual-updates are needed in all these methods to update the angular model and  re-render the react component.

I can streamline this after refactoring the parent components.